Source/Evolve/WarpXEvolve.cpp

```diff
@@ -284,49 +222,7 @@ WarpX::Evolve (int numsteps)
             }
         }
 
-        mypc->ContinuousFluxInjection(cur_time, dt[0]);
-
-        mypc->ApplyBoundaryConditions();
-
-        // interact the particles with EB walls (if present)
-#ifdef AMREX_USE_EB
-        mypc->ScrapeParticles(amrex::GetVecOfConstPtrs(m_distance_to_eb));
-#endif
-
-        m_particle_boundary_buffer->gatherParticles(*mypc, amrex::GetVecOfConstPtrs(m_distance_to_eb));
-
-        // Non-Maxwell solver: particles can move by an arbitrary number of cells
-        if( electromagnetic_solver_id == ElectromagneticSolverAlgo::None ||
-            electromagnetic_solver_id == ElectromagneticSolverAlgo::HybridPIC )
-        {
-            mypc->Redistribute();
-        }
-        else
-        {
-            // Electromagnetic solver: due to CFL condition, particles can
-            // only move by one or two cells per time step
-            if (max_level == 0) {
-                int num_redistribute_ghost = num_moved;
-                if ((m_v_galilean[0]!=0) or (m_v_galilean[1]!=0) or (m_v_galilean[2]!=0)) {
-                    // Galilean algorithm ; particles can move by up to 2 cells
-                    num_redistribute_ghost += 2;
-                } else {
-                    // Standard algorithm ; particles can move by up to 1 cell
-                    num_redistribute_ghost += 1;
-                }
-                mypc->RedistributeLocal(num_redistribute_ghost);
-            }
-            else {
-                mypc->Redistribute();
-            }
-        }
-
-        if (sort_intervals.contains(step+1)) {
```

Comment:
This `if` check could be wrapped into the `mypc->SortParticlesByBin` routine if the sorting related variables are moved to the `MultiParticleContainer` class. That should be done anyway since that is the only place where `sort_interval` and `sort_bin_size` are used.

Reply:
see above.

Could be further substructured into the MPC class (or free standing functions) if we like :)